File: dukevineland.client/src/app/forms/models/add-book-response-dto.ts
import { IReadBook } from '../../shared/models/books/iread-book';

export enum BookReadAddResponseCode {
  Success = 0,
  Duplicate,
  UnknownUser,
  InvalidItem,
  UnknownItem
};

export interface IAddBookResponseDto {

  newItem: IReadBook | null;

  errorCode: number;
  failReason: string;
  userId: string;
}

File: DukeVineland.Dtos/BookEditorDtos/AddBookResponseDto.cs
using DukeVineland.Dtos.TablesDtos;

namespace DukeVineland.Dtos.BookEditorDtos
{
    public enum BookReadAddResponseCode
    {
        Success = 0,
        Duplicate,
        UnknownUser,
        InvalidItem,
        UnknownItem
    };

    public class AddBookResponseDto
    {
        public ReadBook? NewItem { get; set; } = null;

        public int ErrorCode { get; set; } = (int)BookReadAddResponseCode.InvalidItem;

        public string FailReason { get; set; } = string.Empty;

        public string UserId { get; set; } = string.Empty;
    }
}
